### DISCUSSION TITLE
No HDMI Output or Blank Screen

### DISCUSSION THREAD

**Discussion by redbrad0** [raspicamplayer/camplayer#66] [Camplayer OS]
I downloaded and installed this on a clean SD Card. When I try to boot the TV never turns on so I went into config.txt and I uncommented the below so now the TV at least turns on but I still don't get any output which works with a fresh Raspberry install. I have tried uncommenting other settings but havn't been able to get anything to display on the TV. Any advice on where I should look or is there a document on how I can install the Camplayer on my own fresh install?

hdmi_safe=1

**Comment by cybermaus** [comment]
Does the Raspberry show a image when you run a SD card with a normal OS?
Is this a Raspberry Pi 4?

Camplayer does not do anything special to the HDMI, so whatever is the problem, you maybe better off first diagnosing with a normal OS. 

I also has some issues getting it to work, especially on a Raspberry Pi 4. I found the following to be factors:
* Many adapters and cables from micro to normal HDMI loose the EDID signal, so the raspberry goes to 640x480
* Adapter cables also loose signal strength (plays into `config_hdmi_boost` below)
* Modern TV's and monitors do not always support the old 640x480 anymore.
* The raspberry pi 4 does not support the `config_hdmi_boost=x' setting (manual states 'does not support **_yet_**' but its been 5 years....)

So all together, `hdmi_safe=1`, which uses both  640x480 and config_hdmi_boost=7 does not work well on a Pi 4

* You can test if you have EDID signal by running `tvservice -m DMT` and `tvservice -m CEA`, if they produce a list of modes, the EDID works
* You can test if the signal strength is at play, I noticed the monitors stays black, but it also does not go into sleep. Whereas if you then terminate the signal with `tvservice -o` it does go to sleep. If you see that, it means you should be using hdmi_config_boost (cannot on a 4). Or use a straight, short cable, and not a string of conversion cables and pigtails. A signal amplifier, like a active HDMI switch or extention cable, may also work to boost the signal.
* Ignore `hdmi_safe=1` and set to a known working mode for your monitor (if not a RasPi 3, you can also up the boost to 11) :
```
# 2,82=1920x1080, 2,28=1280x800, 47=1440x900
# 2,16=1024x768,  2,23=1280x768,  9=800x600
hdmi_group=2
hdmi_mode=82
config_hdmi_boost=7
```
